### ISSUE TITLE
[IMPROVEMENT] Offer to download/activate maps along GPX track

### ISSUE THREAD

**Issue by scaidermern** [osmandapp/Osmand#2525]
When choosing to navigate along a GPX track or route it would be very helpful if OsmAnd could give a hint when part of the track are outside the region of currently active maps. Just displaying a hint would already be very nice. Ideally, OsmAnd could also offer to download (or activate if already downloaded) missing maps along the route.

**Comment by sonora** [comment]
No further action planned. But please note that you can zoom out to large zooms and visually check which maps are on your device.

**Comment by scaidermern** [comment]
The visual check isn't always easy if the track passes only a few kilometers through a missing country. Of course you can do a visual check to see if the track approaches a border of the installed maps and then do a closer inspection of these track parts to see whether they cross the border.

My assumption was that at least this check would be easy to perform. OsmAnd knows the track points and it knows the border/area of the currently installed maps. Checking both against each other and then showing a message if the track leaves this border into unknown lands.

**Comment by sonora** [comment]
Yes, not extremely hard to implement, but also no main stream request, and no one jumped on it, really. Dev resources are limited, so some focusing is needed every once in a while ...  sorry.

**Comment by scaidermern** [comment]
> Dev resources are limited, so some focusing is needed every once in a while
I can perfectly understand that :)
